docs/ocean/design_docs/vert-lagrang-remap/vert-lagrang-remap.rst

```diff
@@ -0,0 +1,524 @@
+z-tilde coordinate choices.
+
+*Other modifications to ocn_vert_transport_velocity_top are not yet determined.*
+We will likely need to bypass the :code:`ocn_ALE_thickness` call in 
+:code:`ocn_vert_transport_velocity_top` so that the adjustments of layer 
+thickness for SSH perturbations occur during the regridding step.
+
+
+Regridding step(s):
+
+#. :math:`z_k^1`, the depth of the top of the layer, is determined based on 
+   an analytical expression for the grid. 
+   The simplest case is constant z-levels, :math:`z_k^1 = z_k^{init}`.
+   Since :math:`z_k^1` can be a function of the ocean state (e.g., :math:`\rho` 
+   for isopycnal coordinates, regridding doesn't begin until after the solution 
+   for prognostic variables.
+#. Superimpose SSH perturbations according to one of the existing depth-
+   dependent functions, :math:`z_k^2 = z_k^1 + c(z) \: \eta`. As in 
+   :code:`ocn_ALE_thickness`, layer thicknesses are adjusted from the seafloor 
+   upwards. Ideally, there is a single function that is used for both ALE
+   implementations, with and without VLR.
+#. Apply conditioning steps outlined in the following section.
+#. Update :code:`layerThicknessEdge` from the updated :code:`layerThickness`
+   using routine :code:`ocn_diagnostic_solve_layerThicknessEdge`. Note that 
+   :code:`zTop, zMid` are updated later when :code:`ocn_diagnostic_solve` is 
+   called.
+
+All of the regridding steps will be performed from a separate module.
+This topic is further addressed in section Implementation: modularity.
+
+
+Remapping step:
+
+Layer thickness has already been updated to reflect Lagrangian motion when 
+:code:`ocn_tend_thick` is called. This is stored in 
+:code:`layerThickness(tlev=2)`.
+
+There is a new remapping routine with inputs:
+
+- :code:`layerThickness(tlev=2)` for scalars
+- :code:`layerThicknessEdge(tlev=2)` for velocity
+- :code:`bottomDepth`
+- :code:`layerThicknessTarget`
+- :code:`layerThicknessEdgeTarget`
+- Remapped and updated: :code:`normalVelocity`
+- Remapped and updated: All members of :code:`tracerPool` unless 
+  :code:`activeTracersOnly`, in which case only the :code:`activeTracers`
+- *This may not be a complete list*
+
+This routine makes calls to the PPR library
+
+*More details here*
+
+Some implementation considerations for PPR:
+ 
+- Error-checking in PPR: make consistent with MPAS errors, consider additional
+  error checks
+- *Add more here*
+
+:code:`layerThickness(tlev=2)` is then overwritten with the new layer thickness.
+
+After remapping, :code:`ocn_diagnostic_solve` is called. This is needed to 
+compute the density and pressure fields based on the remapped ocean state and
+the diagnostic field :code:`vertVelocityTop` which is the vertical velocity 
+through the top of the layer. This is only used as a diagnostic variable for 
+computing the MOC streamfunction. None of the mixing parameterizations require
+a vertical velocity (Eulerian or diasurface velocity).
```

Comment:
If this is the only usage, it would be worth discussing with @vanroekel, @milenaveneziani and @mark-petersen whether the MOC itself should be computing the this variable as needed when we're in VLR mode.

One consideration is that the MOC also really needs to be computed on an Eulerian (fixed-in-time) or isopicnal grid.  We have treated z-star as sufficiently Eulerian up to now but more complex ALE coordinates would not be.  The vertical velocity through ALE surfaces is not what is needed to compute an accurate MOC.  Instead, the MOC would want to compute the Eulerian vertical velocity (`w = - \nabla \cdot (h_k \mathbf{u}_k)` if I'm not mistaken).  This work is outside the scope of the current design doc but might be worth mentioning as a `.. note::`.